src/tests/ftest/util/server_utils.py

```diff
@@ -1052,3 +655,436 @@
+        elif "response" not in self.information[key]:
+            msg = "No dmg {} scan 'response': status={}".format(
+                key, self.information[key]["status"])
+        elif section not in self.information[key]["response"]:
+            msg = "No '{}' entry found in information 'response': {}".format(
+                section, self.information[key]["response"])
+        if msg:
+            self.log.error(msg)
+            raise ServerFailed("ServerInformation: {}".format(msg))
+
+    def get_numa_node_info(self):
+        """Determine what NVMe and SCM storage is available per socket.
+
+        Raises:
+            ServerFailed: if output from the dmg storage scan is missing or
+                not in the expected format
+
+        Returns:
+            dict: [description]
+
+        """
+        self._check_information("storage", "HostStorage")
+        self._check_information("network", "HostFabrics")
+
+        data = {}
+        try:
+            _info = self.information["storage"]["response"]["HostStorage"]
+            for entry in _info.values():
+                hosts = entry["hosts"].split(":")[0]
+                data[hosts] = {}
+                if entry["storage"]["nvme_devices"]:
+                    data[hosts] = {"nvme": {}}
+                    for device in entry["storage"]["nvme_devices"]:
+                        data[hosts]["nvme"][device["pci_addr"]] = \
+                            device["socket_id"]
+                if entry["storage"]["scm_namespaces"]:
+                    data[hosts] = {"scm": {}}
+                    for device in entry["storage"]["scm_namespaces"]:
+                        data[hosts]["scm"][device["blockdev"]] = \
+                            device["numa_node"]
+        except KeyError as error:
+            raise ServerFailed(
+                "ServerInformation: Error obtaining storage data: {}".format(
+                    error))
+
+        return data
+
+    def get_interface_providers(self):
+        """Determine what interfaces are available with each provider.
+
+        Raises:
+            ServerFailed: if output from the dmg network scan is missing or
+                not in the expected format
+
+        Returns:
+            [type]: [description]
+
+        """
+        self._check_information("network", "HostFabrics")
+
+        data = {}
+        try:
+            _info = self.information["network"]["response"]["HostFabrics"]
+            for entry in _info.values():
+                hosts = entry["HostSet"].split(":")[0]
+                data[hosts] = {}
+                if entry["HostFabric"]["Interfaces"]:
+                    for device in entry["HostFabric"]["Interfaces"]:
+                        if device["Device"] not in data[hosts]:
+                            data[hosts][device["Device"]] = []
+                        data[hosts][device["Device"]].append(device["Provider"])
+        except KeyError as error:
+            raise ServerFailed(
```

Comment:
(pylint-raise-missing-from) Consider explicitly re-raising using the 'from' keyword